src/python/WMCore/Database/DBTest.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python
+#-*- coding: utf-8 -*-
+#pylint: disable=
+"""
+File       : DBTest.py
+Author     : Valentin Kuznetsov <vkuznet AT gmail dot com>
+Description: Base class for WMCore database related unit tests.
+             It defines test database name based on user preferences specified via
+             environment variables and setup logic to setup/tear down actions
+             based on database.
+"""
+from __future__ import print_function
+
+# system modules
+import os
+import sys
+import unittest
+
+class DBTest(unittest.TestCase):
+    def __init__(self, methodName='runTest'):
+        super(WMCoreUnitTest, self).__init__(methodName)
+        self.dbName = os.environ.get('WMCORE_TEST_DATABASE', \
+                'unittest_%s' % self.__class__.__name__)
+        print("WMCoreTestDB test %s database" % self.dbName)
+        enforce = os.environ.get('WMCORE_TEST_DATABASE_DELETE', False)
+        if  enforce:
```

Comment:
Yes, dbName would be either what user provides or unittest_bla.
User can provide additional environment to enforce deletion of database.

On  0, Eric Vaandering <notifications@github.com> wrote:
> > +"""
> > +from __future__ import print_function
> > +
> > +# system modules
> > +import os
> > +import sys
> > +import unittest
> > +
> > +class DBTest(unittest.TestCase):
> > +    def __init__(self, methodName='runTest'):
> > +        super(WMCoreUnitTest, self).__init__(methodName)
> > +        self.dbName = os.environ.get('WMCORE_TEST_DATABASE', \
> > +                'unittest_%s' % self.__class__.__name__)
> > +        print("WMCoreTestDB test %s database" % self.dbName)
> > +        enforce = os.environ.get('WMCORE_TEST_DATABASE_DELETE', False)
> > +        if  enforce:
> 
> Ok, so I think I understand this point now. This is used to crop and recreate the database once per task iff person running it is setting the database name themselves (line 22) and has this flag set.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/dmwm/WMCore/pull/6283/files#r40341166